### PR TITLE
win_service module present/absent

### PR DESCRIPTION
**Pull request by dario-hd**
It's now possible to install/uninstall a windows service using the "present" and the "absent" states. Obviously, to install the service the path to the executable file is mandatory. Also added the optional parameters "display_name" and "description", which are used only when the state is "present".

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @dario-hd. @chrishoffman please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: windows/win_service.ps1 [hunk 1]
```diff
@@ -45,7 +48,29 @@ If ($startMode) {
...
+        Set-Attr $result "changed" $true
+        Set-Attr $result "state" $state
+    } ElseIf ($state -eq "absent") {
+	Exit-Json $result
+    } Else {
+	Fail-Json $result "Service '$svcName' not installed"
+    }
+} Else {
+    If ($state -eq "absent") {
+        Stop-Service -Name $svcName
```
Comment (jhawkesworth) [inline windows/win_service.ps1]: Would it be possible to add some error handling here?
If the stop fails this code will attempt to delete the service and I think you are likely to get 'service marked for deletion' state if this happens.  In my experience the only way to clear a 'service marked for deletion' is to reboot the windows machine.

Comment (dario-hd) [inline windows/win_service.ps1]: Good point, thanks.

**Comment by gregdek** [comment]
@chrishoffman This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by chrishoffman** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @dario-hd for this PR, and thanks @chrishoffman for reviewing. Marking for inclusion.

File: windows/win_service.py [hunk 2]
```diff
@@ -35,6 +35,24 @@
...
+      - The Display Name of the service - Used during service creation
+    required: false
+    default: null
+    aliases: []
+  description:
+    description:
+      - Set the description of the service - Used during service creation
+    required: false
+    default: null
+    aliases: []
```
Comment (bcoca) [inline windows/win_service.py]: empty aliases not needeed

File: windows/win_service.py [hunk 3]
```diff
@@ -35,6 +35,24 @@
...
+  path:
```
Comment (bcoca) [inline windows/win_service.py:49]: new options missing version_added

File: windows/win_service.py [hunk 4]
```diff
@@ -45,11 +63,13 @@
       - disabled
   state:
     description:
-      - C(started)/C(stopped) are idempotent actions that will not run
+      - C(present)/C(absent)/C(started)/C(stopped) are idempotent actions that will not run
```
Comment (bcoca) [inline windows/win_service.py:65]: should make note that new options are only available on versinos 2.1 and above

**Comment by gregdek** [comment]
@dario-hd A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by gregdek** [comment]
@dario-hd Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If we don't hear from you within another 14 days, we will close this pull request.

**Comment by gregdek** [comment]
Looking closer, looks like @bcoca's issues have been addressed (since diffs are outdated). Putting back into shipit.

**Comment by jhawkesworth** [comment]
I realize now this is flagged for shipping but as @nitzmahone has pointed out there is cross over with https://github.com/ansible/ansible-modules-extras/pull/1115 

@dario-hd please can you see if you agree with my most recent comment on https://github.com/ansible/ansible-modules-extras/pull/1115 - I favour having 2 separate modules for this but maybe you have a good counter argument?

**Comment by dario-hd** [comment]
@jhawkesworth I had a look at the other PR and it totally makes sense. I guess this can be closed.

**Comment by bcoca** [comment]
closing as per above

**Pull request closed without merging**